### PR TITLE
De 1166 Allow users to manually select index file(s) for genome browsers.

### PR DESCRIPTION
**Pull request by sri2k1us**
- Also renamed `EnsemblUtil` to `GenomeBrowserUtil` because Ensembl is a genome browser by itself and this class is a generic util.

### CODE REVIEW

File: de-lib/src/main/java/org/iplantc/de/theme/base/client/fileViewers/FileViewerErrorStrings.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
 analysisFailureWarning = Your input contains spaces and/or special characters ({0}) in the path. Please replace all spaces and special characters with an underscore ( _ ) or a dash ( - ).
-indexFileMissingError =  Index file (.bai for .bam (or) .vci for .vcf) is missing. <b>Note:</b> Both genome and index file must be present in the same folder.
+indexFileMissingError =  Unable to locate index file. <b>Note:</b> Both genome and index file must be preset in the same folder.
```
Comment (ash3rz) [inline de-lib/src/main/java/org/iplantc/de/theme/base/client/fileViewers/FileViewerErrorStrings.properties]: Looks like this accidentally got changed from "present" to "preset".

File: de-lib/src/main/java/org/iplantc/de/client/models/errors/diskResources/ErrorGetStat.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org.iplantc.de.client.models.errors.diskResources;
+
+import org.iplantc.de.client.models.HasPaths;
+
+/**
+ * Created by sriram on 10/5/17.
```
Comment (ash3rz) [inline de-lib/src/main/java/org/iplantc/de/client/models/errors/diskResources/ErrorGetStat.java]: Could we add some docs on what this is?

File: de-lib/src/main/java/org/iplantc/de/client/models/errors/diskResources/categories/ErrorGetStatCategory.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import org.iplantc.de.client.models.errors.diskResources.ErrorGetStat;
+import org.iplantc.de.client.util.DiskResourceUtil;
+
+import com.google.common.base.Joiner;
+import com.google.gwt.safehtml.shared.SafeHtml;
+import com.google.web.bindery.autobean.shared.AutoBean;
+
+/**
+ * Created by sriram on 10/5/17.
```
Comment (ash3rz) [inline de-lib/src/main/java/org/iplantc/de/client/models/errors/diskResources/categories/ErrorGetStatCategory.java]: Could we add some docs on what this is?

File: de-lib/src/main/java/org/iplantc/de/fileViewers/client/callbacks/GenomeBrowserUtil.java [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    @Inject
+    DiskResourceUtil diskResourceUtil;
+    @Inject
+    AsyncProviderWrapper<FileSelectDialog> dialogAsyncProviderWrapper;
+    @Inject
+    DiskResourceServiceFacade diskResourceServiceFacade;
+
+    @Inject
+    public GenomeBrowserUtil() {
+
```
Comment (ash3rz) [inline de-lib/src/main/java/org/iplantc/de/fileViewers/client/callbacks/GenomeBrowserUtil.java]: Nice refactor for injection!

File: de-lib/src/main/java/org/iplantc/de/fileViewers/client/callbacks/GenomeBrowserUtil.java [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+            }
+        });
+    }
+
+    protected void processError(Throwable caught,
+                                List<DiskResource> resourcesToSend,
+                                IsMaskable container) {
+        ErrorGetStat egs = AutoBeanCodex.decode(factory, ErrorGetStat.class, caught.getMessage()).as();
+        IPlantDialog dialog = new IPlantDialog();
+        dialog.setHeading(SafeHtmlUtils.fromTrustedString(appearance.indexFileMissing()));
```
Comment (ash3rz) [inline de-lib/src/main/java/org/iplantc/de/fileViewers/client/callbacks/GenomeBrowserUtil.java]: Not necessary for this PR, but ideally it'd be nice to refactor this dialog out into its own class so we can use `AsyncProviderWrapper`, add some static IDs, etc.

Comment (sri2k1us) [reply de-lib/src/main/java/org/iplantc/de/fileViewers/client/callbacks/GenomeBrowserUtil.java]: 👍

**Comment by sri2k1us** [comment]
Thanks for the reviews!

**Pull request merged**